Allow dots in the names, bug #95173.  Patch by Kerin Millar.

diff -urN debianutils-2.14.1/run-parts.8 debianutils-2.14.1-r1/run-parts.8
--- debianutils-2.14.1/run-parts.8	2004-12-10 17:10:37.000000000 +0000
+++ debianutils-2.14.1-r1/run-parts.8	2005-07-12 00:47:44.000000000 +0100
@@ -26,8 +26,8 @@
 Other files and directories are silently ignored.
 
 If the \-\-lsbsysinit option is not given then the names must consist
-entirely of upper and lower case letters, digits, underscores, and
-hyphens.                           
+entirely of upper and lower case letters, digits, underscores, hyphens
+and periods. However, the name must not begin with a period. 
                                                                                 
 If the \-\-lsbsysinit option is given, then the names must not end
 in .dpkg\-old  or .dpkg\-dist or .dpkg\-new or .dpkg\-tmp, and must
diff -urN debianutils-2.14.1/run-parts.c debianutils-2.14.1-r1/run-parts.c
--- debianutils-2.14.1/run-parts.c	2005-02-20 04:25:49.000000000 +0000
+++ debianutils-2.14.1-r1/run-parts.c	2005-07-12 00:47:25.000000000 +0100
@@ -146,7 +148,7 @@
     return !regexec(&tradre, c, 0, NULL, 0);
   }
   else {
-    if (regcomp(&classicalre, "^[a-zA-Z0-9_-]+$", REG_EXTENDED | REG_NOSUB)) {
+    if (regcomp(&classicalre, "^[a-zA-Z0-9_-][a-zA-Z0-9._-]+$", REG_EXTENDED | REG_NOSUB)) {
       error("regex failure");
       exit(1);
     }
